quickjs/src/main/jni/JsMethodProxy.cpp

```diff
@@ -18,53 +18,119 @@
 #include <algorithm>
 #include "Context.h"
 
-JsMethodProxy::JsMethodProxy(JNIEnv *env, const char *name, jobject method)
-    : name(name), methodId(env->FromReflectedMethod(method)) {
+namespace {
+
+std::string getName(JNIEnv *env, jclass javaClass) {
+  const jmethodID method =
+      env->GetMethodID(env->GetObjectClass(javaClass), "getName", "()Ljava/lang/String;");
```

Comment:
As this function is likely to get called quite often in the future, it would be a also a good idea to release the local ref returned by GetObjectClass(), too.

Reply:
Good idea